src/models.rs

```diff
@@ -63,3 +63,25 @@ pub struct NewApiToken {
     pub token: Vec<u8>,
     pub expires_at: Option<SystemTime>,
 }
+
+#[derive(Queryable, Selectable, Debug, Clone)]
+#[diesel(table_name = crate::schema::uploads)]
+#[diesel(check_for_backend(diesel::pg::Pg))]
+pub struct Upload {
+    pub id: Uuid,
+    pub source_code_ipfs_hash: String,
+    pub forc_version: String,
+    pub abi_ipfs_hash: Option<String>,
+    pub bytecode_identifier: Option<String>,
+    pub created_at: SystemTime,
+}
+
+#[derive(Insertable, Debug)]
+#[diesel(table_name = crate::schema::uploads)]
+pub struct NewUpload {
+    pub id: Uuid,
+    pub source_code_ipfs_hash: String,
+    pub forc_version: String,
+    pub abi_ipfs_hash: Option<String>,
+    pub bytecode_identifier: Option<String>,
+}
```

Comment:
Maybe it's fine but my instinct here would be to try and only have 1 of these types considering they are duplicates apart from the final `created_at` field. Disregard if it doesn't make sense in the design though.

Reply:
This is a pattern for working with diesel. Since the created_at field is generated by the database, it's not part of the input when creating a new row entry.